canonical-web-and-design/maas-ui
# Build KVM configuration form.

## Done

- Added KVM configuration form.
- Changed FormikForm buttons so that "Cancel" button is only displayed if `onCancel` prop is provided.
- Drive by fix for refresh form in KVM details so that form closes when successfully completed.
- Drive by fix for TagSelector spacing

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to /MAAS/r/kvm and click on a KVM
- Click on the "Configuration" tab and check that the config form shows up
- Check that you can change and save the values, and the saved values persist

## Fixes

Fixes canonical-web-and-design/MAAS-squad#1960
Fixes canonical-web-and-design/MAAS-squad#2039


## Screenshot
![0 0 0 0_8400_MAAS_r_kvm_180_edit (1)](https://user-images.githubusercontent.com/25733845/87118501-ec2a6e80-c2be-11ea-8e5e-a89a9f853b62.png)
